### ISSUE TITLE
Tree View does not update automatically

### ISSUE THREAD

**Issue by olivercrask** [dendronhq/dendron#367]
**Description**:
When refactoring my hierarchy, the tree view does not update to reflect these changes. Manually running `Dendron: Reload Index` did seem to fix this, though it did leave stubs from the old hierarchy. (Second point potentially related to #224 )

**To reproduce**:

1. Create a hierarchy `bar.foobar` and another named `foo`.

2. Run `Refactor hierarchy` and request that `bar` is changed to `foo.bar`.

3. Look at tree view and no change is observed, even though the file view shows that the refactoring took place

**Expected behaviour**:
That the tree view updates when a refactor occurs and the original hierarchy is not left behind.

**Comment by kevins8** [comment]
Thanks Oli. We're releasing some updates to the tree view with this week's
release that should fix these issues. Stay tuned :)

**Comment by cscorley** [comment]
I am observing a similar issue, that may or may not be related.

In my scenario, I changed from daily journal format from `Y.MMM.Y-MM-DD` back to the default `Y.MM.DD` and started a new journal for today.  Later, I refactored the current tree from `nov` to `11` (see screenshot).  After this refactor, I now have two `11` nodes, a stub for my new journal and an existing one for my old `nov` notes.  This isn't a big deal, since either reload index or reloading VS Code removes the duplicate node.

![image](https://user-images.githubusercontent.com/112752/100643153-afabed80-3307-11eb-9c31-937bc433de79.png)

After the merge occurs, it seems the note now has `stub: true` set in the file.  Removing stub shows the correct state again.
